File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.rbegin(), citations.rend());
    
    int max = citations.front();
    for (int i = max; i >= 0; --i)
    {
        
        bool flag = true;
        
        for(int j = 0; j < citations.size(); ++j)
        {
            if (citations[j] < i)
            {
                if (j >= i)
                    return i;
                else
                    break;
            }
            
            if (j == citations.size() - 1 && j+1 >= i)
                return i;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        for (int i = citations.length; i >= 1; i--) {
            if (isValid(citations, i)) return i;
        }
        return 0;
    }
    
    public boolean isValid(int[] citations, int h) {
        int idx = citations.length - h;
        return citations[idx] >= h;
    }
}